src/api/graph/graph.service.ts

```diff
@@ -0,0 +1,420 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Token } from '@prisma/client';
+import BigNumber from 'bignumber.js';
+
+import { OrderQueryEnum } from '@/api/api.model';
+import { Graph, GraphType, TimeFrame } from '@/api/graph/graph.model';
+import { PairLiquidityInfoHistoryWithTokens } from '@/api/pair-liquidity-info-history/pair-liquidity-info-history.model';
+import { PairLiquidityInfoHistoryService } from '@/api/pair-liquidity-info-history/pair-liquidity-info-history.service';
+import { ContractAddress } from '@/clients/sdk-client.model';
+import { PairDbService } from '@/database/pair/pair-db.service';
+import { PairLiquidityInfoHistoryDbService } from '@/database/pair-liquidity-info-history/pair-liquidity-info-history-db.service';
+import { TokenDbService } from '@/database/token/token-db.service';
+
+const TIME_FRAMES = {
+  '1H': 1,
+  '1D': 24,
+  '1W': 24 * 7,
+  '1M': 24 * 30,
+  '1Y': 24 * 365,
+  MAX: Infinity,
+};
+
+@Injectable()
+export class GraphService {
+  constructor(
+    private readonly pairLiquidityInfoHistoryDb: PairLiquidityInfoHistoryDbService,
+    private readonly tokenDb: TokenDbService,
+    private readonly pairDb: PairDbService,
+    private readonly pairLiquidityInfoHistoryService: PairLiquidityInfoHistoryService,
+  ) {}
+
+  async getGraph(
+    graphType: GraphType,
+    timeFrame: TimeFrame,
+    tokenAddress?: ContractAddress,
+    pairAddress?: ContractAddress,
+  ): Promise<Graph> {
+    let token: Token | undefined = undefined;
+    // VALIDATION
+    if (pairAddress && tokenAddress) {
+      throw new BadRequestException(
+        'It is not possible to request a graph for both tokenAddress and pairAddress',
+      );
+    }
+    if (tokenAddress) {
+      const res = await this.tokenDb.getWithAggregation(tokenAddress);
+      if (!res) {
+        throw new NotFoundException('Token not found');
+      }
+      token = res;
+    }
+    if (pairAddress) {
+      const pair = await this.pairDb.getOne(pairAddress);
+      if (!pair) {
+        throw new NotFoundException('Pair not found');
+      }
+    }
+
+    const history = await this.pairLiquidityInfoHistoryDb.getAll({
+      limit: 9999999,
```

Comment:
can we remove this since we are now calling this internally?

Reply:
✅